src/main/ruby/truffleruby/core/splitter.rb

```diff
@@ -95,24 +95,24 @@ def valid_encoding?(string)
         raise ArgumentError, "invalid byte sequence in #{string.encoding.name}" unless string.valid_encoding?
       end
 
-      def split_type_chars(string, limit, block)
+      def split_type_chars(string, limit, callable)
         if limit > 0
           last = string.size > (limit - 1) ? string[(limit - 1)..-1] : empty_string(string)
 
           string.each_char.each_with_index do |char, index|
             break if index == limit - 1
-            block.call(char)
+            callable.call(char)
           end
 
-          block.call(last)
+          callable.call(last)
         else
-          string.each_char(&block)
+          string.each_char(&callable)
```

Comment:
This will turn it into a `Proc` though

Reply:
A simple fix for now, to avoid relying on Method#to_proc, would be
```suggestion
          string.each_char { |char| callable.call(char) }
```

Reply:
@andrykonchin Can you do this since you already have a commit on top?